pkg/ebpf/c/common/context.h

```diff
@@ -215,6 +218,33 @@ statfunc int init_program_data(program_data_t *p, void *ctx)
 }
 // clang-format on
 
+statfunc int setup_policies(program_data_t *p)
+{
+    if (p->event->context.policies_version)
+        return 0; // already set
+
+    u16 version = p->config->policies_version;
+    policies_config_map_t *policies_config_map;
+    policies_config_map = bpf_map_lookup_elem(&policies_config_version, &version);
+    if (unlikely(policies_config_map == NULL))
+        return 1;
+
+    u32 zero = 0;
+    policies_config_t *policies_config;
+    policies_config = bpf_map_lookup_elem(policies_config_map, &zero);
+    if (unlikely(policies_config == NULL))
+        return 1;
+
+    // copy policies_config to event
+    if (bpf_probe_read(&p->event->policies_config, sizeof(policies_config_t), policies_config) != 0)
+        return 1;
```

Comment:
Why do we need all of this?
Can't we just use p->config->policies_config like we did in get_policies_config()?

Reply:
In `get_policies_config()` we were not caching the policies_config, only short-circuiting to use the one in p->config, but if a different version comes, we were using the pointer of the map directly.

Now, we're caching it in `p->event->policies_config` avoiding lookups.

Reply:
And we can't rely on `p->config->policies_config` since it can change meanwhile (new versions).

Reply:
This was solved in the last pushes. Thanks.